com.unity.ml-agents/Runtime/Actuators/ActionSpaceDef.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.MLAgents.Policies;
+
+namespace Unity.MLAgents.Actuators
+{
+    public readonly struct ActionSpaceDef
+    {
+
+        public readonly SpaceType SpaceType;
+
+        /// <summary>
+        /// An array of branch sizes for our action space.
+        ///
+        /// For an IActuator that uses a Discrete <see cref="SpaceType"/>, the number of
+        /// branches is the Length of the Array and each index contains the branch size.
+        /// The cumulative sum of the total number of discrete actions can be retrieved
+        /// by the <see cref="sumOfDiscreteBrancheSizes"/> property.
+        ///
+        /// For an IActuator with a Continuous <see cref="SpaceType"/>, the number of continuous
+        /// actions is at the 0th index of array.
+        /// </summary>
+        public readonly int[] BranchSizes;
+
+        /// <summary>
+        /// The number of branches for a Discrete <see cref="SpaceType"/>, or the number of actions
+        /// for a Continuous <see cref="SpaceType"/>.
+        /// </summary>
+        public readonly int NumActions;
+
+        /// <summary>
+        /// Get the total number of Discrete Actions that can be taken by calculating the Sum
+        /// of all of the Discrete Action branch sizes.
+        /// </summary>
+        public int sumOfDiscreteBrancheSizes { get; }
```

Comment:
nit: `SumOf...` (https://ono.unity3d.com/unity-extra/unity-meta/files/@/ReferenceSource/CSharp/Assets/CSharpReference.cs#L193 "In the Unity namespace, public properties are PascalCase with no prefix")